sdk/storage/Azure.Storage.Blobs/src/BlobQuickQueryStream.cs

```diff
@@ -131,6 +131,14 @@ private async Task<int> ReadInternal(bool async, byte[] buffer, int offset, int
                     case Constants.QuickQuery.DataRecordName:
                         record.TryGetValue(Constants.QuickQuery.Data, out object byteObject);
                         byte[] bytes = (byte[])byteObject;
+
+                        if (_buffer != null)
+                        {
+                            ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
Always set `clearArray = true` when there's customer data

Reply:
Nit - we could check whether the current buffer size is larger than `bytes.Length` and not bother returning/renting a new one.

Reply:
Fixed in next iteration.